napari/components/viewer_model.py

```diff
@@ -395,9 +399,15 @@ def reset_view(
         scene_size = extent[1] - extent[0]
         corner = extent[0]
         grid_size = list(self.grid.actual_shape(len(self.layers)))
+
         if len(scene_size) > len(grid_size):
             grid_size = [1] * (len(scene_size) - len(grid_size)) + grid_size
-        size = np.multiply(scene_size, grid_size)
+
+        # total spacing accounts for the distance between layers, results in 0 if not grid mode (grid_size = [1, 1] - 1)
+        total_spacing = np.multiply(
+            scene_size, self.grid.spacing * (np.array(grid_size) - 1)
+        )
+        size = np.multiply(scene_size, grid_size) + total_spacing
```

Comment:
Noted in the community meeting:
Currently if you have a non-square layer, you will get uneven spacing between the layers.
Probably best to have a consistent "border" zone computed from just one of the dimensions -- the smallest?